django_netjsonconfig/base/admin.py

```diff
@@ -200,6 +205,12 @@ def download_view(self, request, pk):
         return send_file(filename='{0}.tar.gz'.format(config.name),
                          contents=config_archive.getvalue())
 
+    def device_context_view(self, request, pk):
+        device = get_object_or_404(self.model, pk=pk)
+        device_config = Config(device=device)
```

Comment:
you should do this only when the device does not have a config.

But actually, I think `get_context` being implemented at `Config` level does not have sense anymore and we should fix that.

Can you move and adapt the `get_context` method to the `Device` model and keep a backward compatible `Config.get_context` which calls `self.device.get_context()`?

That way you will be able to simplify this code to just return `device.get_context()`

Reply:
Okay, I made changes do they look good?